tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs

```diff
@@ -122,7 +173,68 @@ public async Task SubmitsTraces(string path, HttpStatusCode statusCode)
                 statusCode = (HttpStatusCode)500;
             }
 
-            var spans = await GetWebServerSpans(path, _iisFixture.Agent, _iisFixture.HttpPort, statusCode);
+            // Append virtual directory to the actual request
+            var spans = await GetWebServerSpans(_iisFixture.VirtualApplicationPath + path, _iisFixture.Agent, _iisFixture.HttpPort, statusCode);
+
+            var sanitisedPath = VerifyHelper.SanitisePathsForVerify(path);
+
+            var settings = VerifyHelper.GetSpanVerifierSettings(sanitisedPath, (int)statusCode);
+
+            // Overriding the type name here as we have multiple test classes in the file
+            // Ensures that we get nice file nesting in Solution Explorer
+            await Verifier.Verify(spans, settings)
+                          .UseMethodName("_")
+                          .UseTypeName(_testName);
+        }
+    }
+
+    [UsesVerify]
+    public abstract class AspNetMvc5ModuleOnlyTests : TestHelper, IClassFixture<IisFixture>
+    {
+        private readonly IisFixture _iisFixture;
+        private readonly string _testName;
+
+        protected AspNetMvc5ModuleOnlyTests(IisFixture iisFixture, ITestOutputHelper output, bool classicMode, bool enableRouteTemplateResourceNames, bool virtualApp = false)
+            : base("AspNetMvc5", @"test\test-applications\aspnet", output)
+        {
+            SetServiceVersion("1.0.0");
+            SetEnvironmentVariable(ConfigurationKeys.FeatureFlags.RouteTemplateResourceNamesEnabled, enableRouteTemplateResourceNames.ToString());
+
+            // Disable the MVC part, so we can't back propagate any details to the tracing module
+            SetEnvironmentVariable(ConfigurationKeys.DisabledIntegrations, nameof(Configuration.IntegrationId.AspNetMvc));
+
+            _iisFixture = iisFixture;
+            _iisFixture.ShutdownPath = "/home/shutdown";
+            if (virtualApp)
+            {
+                _iisFixture.VirtualApplicationPath = "/my-app";
+            }
+
+            _iisFixture.TryStartIis(this, classicMode ? IisAppType.AspNetClassic : IisAppType.AspNetIntegrated);
+            _testName = nameof(AspNetMvc5Tests)
+                      + "ModuleOnly"
+                      + (virtualApp ? ".VirtualApp" : string.Empty)
+                      + (classicMode ? ".Classic" : ".Integrated");
+        }
+
+        public static TheoryData<string, int> Data() => new()
+        {
+            { "/Home/Index", 200 },
+            { "/Home/Get", 500 },
+            { "/badrequest", 500 },
+            { "/statuscode/503", 503 },
+            { "/BadRequestWithStatusCode/401", 500 }, // the exception thrown here overrides the status code set
```

Comment:
The previous tests use the following query strings. Can you replace or update the test cases to use these? This tests the TransferRequest functionality of ASP.NET

```
{ "/BadRequestWithStatusCode/401?TransferRequest=true", 401 },
{ "/BadRequestWithStatusCode/503?TransferRequest=true", 503 },
```

Reply:
Actually, my suggestion may not add much value. The current test you have is helpful for the status code update, but if you don't think the additional tests are worth it feel free to mark as "Won't Fix"

Reply:
IIRC, this was just me being lazy, because you have to faff a bit with transfer request when using classic mode 😄 I've added them back in https://github.com/DataDog/dd-trace-dotnet/pull/2851/commits/9764168b611bb4e59f81a4afa3bb250b5e8f4cc6. AFAICT, they look pretty much as we'd hope for (with the caveat of the additional prefix to the virtual path tests)